receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -0,0 +1,316 @@
+)
+
+type alertsReceiver struct {
+	addr        string
+	secret      string
+	server      *http.Server
+	tlsSettings *configtls.TLSServerSetting
+	consumer    consumer.Logs
+	wg          *sync.WaitGroup
+	logger      *zap.Logger
+}
+
+func newAlertsReceiver(logger *zap.Logger, cfg AlertConfig, consumer consumer.Logs) (*alertsReceiver, error) {
+	var tlsConfig *tls.Config
+
+	if cfg.TLS != nil {
+		var err error
+
+		tlsConfig, err = cfg.TLS.LoadTLSConfig()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	recv := &alertsReceiver{
+		addr:        cfg.Endpoint,
+		secret:      cfg.Secret,
+		tlsSettings: cfg.TLS,
+		consumer:    consumer,
+		wg:          &sync.WaitGroup{},
+		logger:      logger,
+	}
+
+	s := &http.Server{
+		TLSConfig: tlsConfig,
+		Handler:   http.HandlerFunc(recv.handleRequest),
+	}
+
+	recv.server = s
+
+	return recv, nil
+}
+
+func (a alertsReceiver) Start(ctx context.Context, host component.Host) error {
+	// We use a.server.Serve* over a.server.ListenAndServe*
+	// So that we can catch and return errors relating to binding to network interface on start.
+	var lc net.ListenConfig
+
+	l, err := lc.Listen(ctx, "tcp", a.addr)
+	if err != nil {
+		return err
+	}
+
+	a.wg.Add(1)
+	if a.tlsSettings != nil {
+		go func() {
+			defer a.wg.Done()
+
+			a.logger.Debug("Starting ServeTLS",
+				zap.String("address", a.addr),
+				zap.String("certfile", a.tlsSettings.CertFile),
+				zap.String("keyfile", a.tlsSettings.KeyFile))
+
+			err := a.server.ServeTLS(l, a.tlsSettings.CertFile, a.tlsSettings.KeyFile)
+
+			a.logger.Debug("Serve TLS done")
+
+			if err != http.ErrServerClosed {
+				a.logger.Error("ServeTLS failed", zap.Error(err))
+				host.ReportFatalError(err)
+			}
+		}()
+	} else {
+		go func() {
+			defer a.wg.Done()
+
+			a.logger.Debug("Starting Serve", zap.String("address", a.addr))
+
+			err := a.server.Serve(l)
+
+			a.logger.Debug("Serve done")
+
+			if err != http.ErrServerClosed {
+				a.logger.Error("Serve failed", zap.Error(err))
+				host.ReportFatalError(err)
+			}
+		}()
+	}
+
+	return nil
+}
+
+func (a alertsReceiver) handleRequest(rw http.ResponseWriter, req *http.Request) {
+	if req.ContentLength < 0 {
+		rw.WriteHeader(http.StatusBadRequest)
+		a.logger.Debug("Got request with no Content-Length specified", zap.String("remote", req.RemoteAddr))
+		return
+	}
+
+	if req.ContentLength > maxContentLength {
+		rw.WriteHeader(http.StatusBadRequest)
+		a.logger.Debug("Got request with large Content-Length specified",
+			zap.String("remote", req.RemoteAddr),
+			zap.Int64("content-length", req.ContentLength),
+			zap.Int64("max-content-length", maxContentLength))
+		return
+	}
+
+	payload := make([]byte, req.ContentLength)
+	_, err := io.ReadFull(req.Body, payload)
+	if err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		a.logger.Debug("Failed to read alerts payload", zap.Error(err), zap.String("remote", req.RemoteAddr))
+		return
+	}
+
+	payloadSigHeader := req.Header.Get(signatureHeaderName)
```

Comment:
nit: Might be worth checking the headers before trying to read in the payload. Fail fast mentality.